cli/deno_dir.rs

```diff
@@ -108,228 +176,135 @@ impl DenoDir {
 
     let root: PathBuf = custom_root.unwrap_or(default);
     let gen = root.as_path().join("gen");
+    let gen_cache = DiskCache::new(&gen);
     let deps = root.as_path().join("deps");
-    let deps_http = deps.join("http");
-    let deps_https = deps.join("https");
-
-    // Internally within DenoDir, we use the config as part of the hash to
-    // determine if a file has been transpiled with the same configuration, but
-    // we have borrowed the `State` configuration, which we want to either clone
-    // or create an empty `Vec` which we will use in our hash function.
-    let config = match state_config {
-      Some(config) => config.clone(),
-      _ => b"".to_vec(),
-    };
+    let deps_cache = DiskCache::new(&deps);
 
     let deno_dir = Self {
       root,
-      gen,
-      deps,
-      deps_http,
-      deps_https,
-      config,
+      gen_cache,
+      deps_cache,
       progress,
-      download_cache: DownloadCache::default(),
+      source_file_cache: SourceFileCache::default(),
     };
 
     // TODO Lazily create these directories.
-    deno_fs::mkdir(deno_dir.gen.as_ref(), 0o755, true)?;
-    deno_fs::mkdir(deno_dir.deps.as_ref(), 0o755, true)?;
-    deno_fs::mkdir(deno_dir.deps_http.as_ref(), 0o755, true)?;
-    deno_fs::mkdir(deno_dir.deps_https.as_ref(), 0o755, true)?;
+    // TODO: once saving and loading of SourceFiles uses DiskCache API these calls can be removed
+    deno_fs::mkdir(deno_dir.deps_cache.location.as_ref(), 0o755, true)?;
+    let deps_http = deps.join("http");
+    let deps_https = deps.join("https");
+    deno_fs::mkdir(deps_http.as_ref(), 0o755, true)?;
+    deno_fs::mkdir(deps_https.as_ref(), 0o755, true)?;
 
     debug!("root {}", deno_dir.root.display());
-    debug!("gen {}", deno_dir.gen.display());
-    debug!("deps {}", deno_dir.deps.display());
-    debug!("deps_http {}", deno_dir.deps_http.display());
-    debug!("deps_https {}", deno_dir.deps_https.display());
+    debug!("deps {}", deno_dir.deps_cache.location.display());
+    debug!("gen {}", deno_dir.gen_cache.location.display());
 
     Ok(deno_dir)
   }
 
-  // https://github.com/denoland/deno/blob/golang/deno_dir.go#L32-L35
-  pub fn cache_path(
-    self: &Self,
-    filepath: &Path,
-    source_code: &[u8],
-  ) -> (PathBuf, PathBuf) {
-    let cache_key =
-      source_code_hash(filepath, source_code, version::DENO, &self.config);
-    (
-      self.gen.join(cache_key.to_string() + ".js"),
-      self.gen.join(cache_key.to_string() + ".js.map"),
-    )
+  /// This method returns local file path for given module url that is used
+  /// internally by DenoDir to reference module.
+  ///
+  /// For specifiers starting with `file://` returns the input.
+  ///
+  /// For specifier starting with `http://` and `https://` it returns
+  /// path to DenoDir dependency directory.
+  pub fn url_to_deps_path(self: &Self, url: &Url) -> PathBuf {
```

Comment:
Maybe delegate this task to the deps_cache? (still we could possibly leave a convenience wrapper here)

Reply:
Yep, that makes sense

Reply:
Removed completely